inc/mailcollector.class.php

```diff
@@ -776,16 +807,25 @@ function collect($mailgateID, $display = 0) {
                   }
                   $refused++;
                   $rejected->add($rejinput);
-                  $this->deleteMails($uid, self::REFUSED_FOLDER);
+                  $delete[$uid] =  self::REFUSED_FOLDER;
                }
                $this->fetch_emails++;
             }
-            imap_expunge($this->marubox);
-            $this->closeMailbox();   //Close Mail Box
+
+            krsort($delete);
+            foreach ($delete as $uid => $folder) {
+               $this->deleteMails($uid, $folder);
+            }
 
             //TRANS: %1$d, %2$d, %3$d, %4$d and %5$d are number of messages
-            $msg = sprintf(__('Number of messages: available=%1$d, retrieved=%2$d, refused=%3$d, errors=%4$d, blacklisted=%5$d'),
-                           $tot, $this->fetch_emails, $refused, $error, $blacklisted);
+            $msg = sprintf(
+               __('Number of messages: available=%1$d, retrieved=%2$d, refused=%3$d, errors=%4$d, blacklisted=%5$d'),
+               $this->getTotalMails(),
```

Comment:
As mail count is retrieve on current state of the box, you have to get this count before moving mails elsewhere.